### ISSUE TITLE
✨ [REQUEST] Jellyfish NAS: omit `/config/addons_config/jellyfin/data/metadata` from backup

### ISSUE THREAD

**Issue by bilogic** [alexbelgium/hassio-addons#1048]
### Checked

- [X] If a new addon, I have checked on  Google that such as addon doesn't already exists
- [X] If a new feature, I have searched in the discussions & issues that it was not already covered

### Which addon?

Jellyfish NAS

### Is your feature request related to a problem? Please describe

Yes, the `/config/addons_config/jellyfin/data/metadata` folder is over 2GB and that bulks up my backup which was originally less than 500MB

### Describe the solution you'd like

I like to have the backup as small as possible

**Comment by txitxo0** [comment]
> ### Checked
> 
> - [X] If a new addon, I have checked on  Google that such as addon doesn't already exists
> - [X] If a new feature, I have searched in the discussions & issues that it was not already covered
> 
> ### Which addon?
> 
> Jellyfish NAS
> 
> ### Is your feature request related to a problem? Please describe
> 
> Yes, the `/config/addons_config/jellyfin/data/metadata` folder is over 2GB and that bulks up my backup which was originally less than 500MB
> 
> ### Describe the solution you'd like
> 
> I like to have the backup as small as possible

Why don't you just avoid the jellyfin addon from the backup? Do you need to keep safe all this metadata?

**Comment by bilogic** [comment]
Thanks! Your comment made me relook at the backup add on and there was a field to specify folders to exclude.

**Comment by bilogic** [comment]
I like to backup the jellyfin configuration, but definitely not this metadata. Ideally, it would be great if I could just exclude `/config/addons_config/jellyfin/data/metadata` from the backup.

**Comment by alexbelgium** [comment]
As is said in the thread you’ve linked :

> No, unfortunately that's not possible.
> 
> This addon uses (as most other backup addons) the functionality of the supervisor to create the backups. And the supervisor only allows to exclude the five folders: "config", "ssl", "share", "addons/local" and "media".

alas it is still the same ; the true solution would be to be to request to the HA team to implement something such as a .exclude feature.

because if I store the metadata in /data, it will still take place in backups, and if I store it in a non rémanent folder people will be unhappy as it would rebuild metadata at each start…

**Comment by bilogic** [comment]
`.exclude` sounds like a great idea!

**Comment by alexbelgium** [comment]
Yes but I can’t implement that it must be done in the ha supervisor

**Comment by alexbelgium** [comment]
Hi, I've pushed a test version could you please check ? 

I've moved those folders to /data, which allows to take advantage from the embedded feature from HA to exclude folders stored in /data

**Comment by alexbelgium** [comment]
Typo, I'll push a test2

**Comment by txitxo0** [comment]
Oh! Good catch! Great idea!

**Comment by bilogic** [comment]
What do I need to do? I moved `metadata` to `/data/metadata`, or should it be `/data/data/metadata`?

**Comment by bilogic** [comment]
![image](https://github.com/alexbelgium/hassio-addons/assets/946010/c922ea2b-c1fe-46fb-8389-cecec38208fe)

I manually did this, correct?
Testing the backup now.

**Comment by alexbelgium** [comment]
Oh, everything should have been done automatically ! And your current metadata folder should have moved to /data/data/metadata (not pretty but easiest for me to keep the existing structure)

**Comment by bilogic** [comment]
How about the backup? Should I be excluding certain folders etc?
Actually I did try creating a symlink yesteday. The backup follow the links and was still as big, it did not shrink.

**Comment by bilogic** [comment]
Nope, I'm still getting 1.5GB backup

**Comment by bilogic** [comment]
Wait, correction. It works! Something else got bloated, let me fix on my end. Thank you!

**Comment by bilogic** [comment]
![image](https://github.com/alexbelgium/hassio-addons/assets/946010/f2b7d7a0-1f5f-46ed-a5ad-6b7dd8267690)

After moving `metadata` to `/data/metadata`, it got backed up to vscode instead. Any idea?

**Comment by alexbelgium** [comment]
Well this would be most unlikely… one add on cannot communicate with others data folders. I would recommend that you put back this folder in the initial jellyfin/data/metadata, then restart the jellyfin addon and check that the folder disappears. Thanks !

**Comment by bilogic** [comment]
![image](https://github.com/alexbelgium/hassio-addons/assets/946010/5fb7e814-585d-41c7-9837-c2747817a548)

Ok, I stopped JF and did what you said and got the above using VSCode

After JF restarted, `transcode` became a symlink, but `metadata` stayed as-is

**Comment by alexbelgium** [comment]
Hi, ingress is not working with jellyfin in the latest version. Was it previously ? Thanks

**Comment by bilogic** [comment]
what do you mean by `ingress`?

**Comment by alexbelgium** [comment]
Accessing jellyfin from the sidebar in HA, instead of using ip:port

**Comment by bilogic** [comment]
If you use HTTP, ingress works, but if configured for HTTPS, ingress stops working.
I switch to using caddy to provide HTTPS when accessing jellyfin directly, so it works both HTTP and HTTPS now.

**Comment by alexbelgium** [comment]
Thanks

**Comment by txitxo0** [comment]
I am thinking maybe it is good to use the same strategy in the emby addon! :)

**Comment by bilogic** [comment]
Using caddy also cuts down on the need to to deal with the quirks of each addon individually, e.g. Navidrome needs the cert in PFX format... 🤦‍♂

**Comment by nordeep** [comment]
Seems last changes with folders fully broken metadata in my case. Additionally, excluding metadata doesn't seem like a good solution. Sometimes this metadata is unique, such as custom pictures.

**Comment by alexbelgium** [comment]
Then I'll put back things how they were and think about it again ;-)

**Comment by alexbelgium** [comment]
A way to test without modying addon is to use my custom script mod : https://github.com/alexbelgium/hassio-addons/wiki/Add%E2%80%90ons-feature-:-customisation

To do that, just create the file : /config/addons_autoscript/jellyfin.sh and put inside something like (not tested) : 

```shell
#!/bin/bash
echo "Starting script"
# Create folder if not existing
mkdir -p /data/metadata
# Add permissions to LSIO default user
chown abc:abc /data/metadata
chmod 755 /data/metadata
# Remove folder if existing
if [ -d /config/addons_config/jellyfin/metadata ]; then rm -r /config/addons_config/jellyfin/metadata; fi
ln -s /data/metadata /config/addons_config/jellyfin
echo "Done"
```

**Comment by bilogic** [comment]
@nordeep 

1. I thought that `metadata` is automatically created by JF and acts like a cache? 
2. How are you adding custom data to it? 
3. From what I observe, JF adds some images etc right beside my media folders, `*-backdrop.jpg`, `*-posters.jpg`, `*-logo.jpg`, that's ok

**Comment by nordeep** [comment]
@bilogic 

1. I'm not expert in JF. As I know metadata is updating by request from web-interface. In my case all metadata was gone as a result - no images at all in JF. I checked the folders and saw the reorganization, but the metadata folder was empty. As I understand, the script did not work correctly. Good thing I had a backup. I tried to move restored metadata to the new folder, but without success, the images did not appear in JF.
2. Adding posters via the web interface. Not all posters are available through JF fetch. Some posters I have upload manually.
3. As I know, images are stored in the metadata folder.
4. It's not exactly convenient to restore JF from a backup and lose all the images.

**Comment by alexbelgium** [comment]
Indeed... I've put the code as it was, and then propose to deal with avoiding the backup with my specific script above

Btw, HomeAssistant has once again change the logic of addons and the /config won't be available anymore so it will likely break lots of things...

**Comment by bilogic** [comment]
@alexbelgium what's changed? any links? Thanks.

**Comment by alexbelgium** [comment]
Sorry it was this push https://github.com/alexbelgium/hassio-addons/commit/fbc68bd42892c9801b533b26c39d30b302a80af6 normally the latest version should restore the old metadata behavior

**Comment by nordeep** [comment]
@alexbelgium I didn't update to the latest version jet. Just downgrade to .11 version and restore metadata from backup. Personally, it was not a problem for me.  Thank you for your hard work and the time you spend!

**Comment by bilogic** [comment]
@alexbelgium May I suggest having a JF lite? The difference is only in the config file, correct? If it is too much trouble, don't worry. I will create my own fork. Thank you.

**Comment by bilogic** [comment]
@alexbelgium 

Pardon me, too many things happening on my side to pay close attention.

1. May I ask, will your eventual JF backup or don't backup `metadata`?
2. If it backs up `metadata`, what are my options to not back it up? Create the script https://github.com/alexbelgium/hassio-addons/issues/1048#issuecomment-1807209301 to not back it up?

Thanks!

**Comment by alexbelgium** [comment]
Hi, summary is :
- The current code reverts behavior and backups metadata, as previously
- If you want to avoid backing it up, the best way is to used my proposed wow (not tested) to skip it
- HA has introduced some breaking change where addons don't have access anymore to /config, so it will break things but I'm not sure when it will start happening (not related to this specific thread but general comment)

**Comment by encryptix** [comment]
> Hi, I've pushed a test version could you please check ?
> 
> I've moved those folders to /data, which allows to take advantage from the embedded feature from HA to exclude folders stored in /data

I've followed the instructions above to move the metadata (and cache /config/addons_config/jellyfin/cache) which took it out of the general folder. The data is now included as part of the jellyfin addon, which made it easy to exclude from backups.

Is that how the workaround above is meant to work? Just want to be sure i've it set up correctly and that theres no way of backing up jellyfin too (but not the data).

`which allows to take advantage from the embedded feature from HA to exclude folders stored in /data`
I wasnt too sure what this meant.

Thanks

**Comment by ve6rah** [comment]
The last post on this topic is now over a year old. Any progress on moving Jellyfin's data out of the addons_config folder? This is adding over 2GB to my home assistant backups, and I really don't have the space for that where I'm storing my backups.

It seems that none of my other addons are storing their config in this location, only jellyfin?

**Comment by encryptix** [comment]
After doing the symlink approach above and that working well, it then seemed to cause issues when a large upgrade happened and the web portion didnt update. 
I switched to using https://github.com/mdvorak/ha-addon-jellyfin which says
`
Configuration and caches are stored on share (not in config), therefore its data will not be deleted when add-on is uninstalled, and must be deleted manually. This is intentional
`

I think have another method of backing up this directory, not in the HA backup.

**Comment by ve6rah** [comment]
> After doing the symlink approach above and that working well, it then seemed to cause issues when a large upgrade happened and the web portion didnt update. I switched to using https://github.com/mdvorak/ha-addon-jellyfin which says `Configuration and caches are stored on share (not in config), therefore its data will not be deleted when add-on is uninstalled, and must be deleted manually. This is intentional`
> 
> I think have another method of backing up this directory, not in the HA backup.

Was there a simple way to migrate? Or did you basically have to start over?

**Comment by alexbelgium** [comment]
Hi, actually the other addon has the exact same behavior as mine : it just stores data in /share instead of /config but globally still takes the same amount of backup space (considering you would want to backup your jellyfin database). There are solution to avoid saving the metadata directory but discussion above is that it is a valuable data piece that is anyway part of the emby database. But sure what would be the gain though 

Switching from /config to anything else could be super easy for this addon but the added value wasn't clear from the discussion above

**Comment by alexbelgium** [comment]
Actually the feature is already in place just use the `data_location` option
